src/daft-micropartition/src/micropartition.rs

```diff
@@ -870,7 +870,7 @@ pub fn read_csv_into_micropartition(
             let unioned_schema = tables
                 .iter()
                 .map(|tbl| tbl.schema.clone())
-                .try_reduce(|s1, s2| s1.union(s2.as_ref()).map(Arc::new))?
+                .reduce(|s1, s2| Arc::new(s1.non_distinct_union(s2.as_ref())))
```

Comment:
When do we have two tables that we are unioning that have common columns? Just want to make sure that this non-distinct union is the correct behavior.

Reply:
This really only gets used in the MicroPartition API for reading multiple parquet files. E.g. `daft.table.MicroPartition.read_parquet_bulk(["file1.parquet", "file2.parquet"])`

Here both files can have the same columns.

The other MicroPartition APIs for `read_parquet`, `read_csv`, and `read_json` are non-concerns because they only ever take in one uri. But the `read_{csv, json, parquet}_into_micropartition` functions they call under the hood take in a slice of uris and can run into the same problem that `read_parquet_bulk` currently does. As of today there are no current users that read more than one API.

FWIW I think the original author intended to use the semantics of a non-distinct union. But I'm not 100% sure why we would even bother with the cases where the schemas were mismatched---I imagine this would quickly blow up elsewhere.